cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/helpers.js

```diff
@@ -0,0 +1,342 @@
+) {
+  getExpandedDocRowValue(docTableRowNumber, expandedDocumentRowNumber).then(
+    ($expandedDocumentRowValue) => {
+      const filterFieldText = $expandedDocumentRowValue.text();
+      getExpandedDocRow(docTableRowNumber, expandedDocumentRowNumber).within(() => {
+        cy.getElementByTestId('removeInclusiveFilterButton').click();
+      });
+      // Verify pill text
+      cy.getElementByTestId('globalFilterLabelValue', {
+        timeout: 10000,
+      }).should('have.text', filterFieldText);
+      cy.getElementByTestId('discoverQueryHits').should(
+        'have.text',
+        expectedQueryHitsAfterFilterApplied
+      ); // checkQueryHitText must be in front of checking first line text to give time for DocTable to update.
+      toggleDocTableRow(docTableRowNumber);
+      getExpandedDocRowValue(docTableRowNumber, expandedDocumentRowNumber).should(
+        'not.have.text',
+        filterFieldText
+      );
+    }
+  );
+  cy.getElementByTestId('globalFilterBar').find('[aria-label="Delete"]').click();
+  cy.getElementByTestId('discoverQueryHits').should('have.text', expectedQueryHitsWithoutFilter);
+  toggleDocTableRow(docTableRowNumber);
+}
+
+/**
+ * Check the first expanded Doc Table Field's first row's Toggle Column button has intended behavior.
+ */
+export function verifyDocTableFirstExpandedFieldFirstRowToggleColumnButtonHasIntendedBehavior() {
+  getExpandedDocRowFieldName(0, 0).then(($expandedDocumentRowFieldText) => {
+    const fieldText = $expandedDocumentRowFieldText.text();
+    getExpandedDocRow(0, 0).within(() => {
+      cy.getElementByTestId('docTableHeader-' + fieldText).should('not.exist');
+      cy.getElementByTestId('toggleColumnButton').click();
+    });
+    cy.getElementByTestId('fieldList-selected').within(() => {
+      cy.getElementByTestId('field-' + fieldText).should('exist');
+    });
+    cy.getElementByTestId('docTableHeader-' + fieldText).should('exist');
+    cy.getElementByTestId('fieldToggle-' + fieldText).click();
+    cy.getElementByTestId('fieldList-selected').within(() => {
+      cy.getElementByTestId('field-' + fieldText).should('not.exist');
+    });
+    cy.getElementByTestId('docTableHeader-' + fieldText).should('not.exist');
+  });
+}
+
+/**
+ * Check the first expanded Doc Table Field's first row's Exists Filter button filters the correct Field.
+ * @param {number} docTableRowNumber Integer starts from 0 for the first row
+ * @param {number} expandedDocumentRowNumber Integer starts from 0 for the first row
+ * @param {string} expectedQueryHitsWithoutFilter expected number of hits in string after the filter is removed Note you should add commas when necessary e.g. 9,999
+ * @param {string} expectedQueryHitsAfterFilterApplied expected number of hits in string after the filter is applied. Note you should add commas when necessary e.g. 9,999
+ */
+export function verifyDocTableFirstExpandedFieldFirstRowExistsFilterButtonFiltersCorrectField(
+  docTableRowNumber,
```

Comment:
why we need a specific utility for the first row?

Reply:
Moving it to the spec instead.